core/common/test/InstantTest.kt

```diff
@@ -128,7 +128,16 @@ class InstantTest {
             "-04",
         )
 
-        val offsets = offsetStrings.map { UtcOffset.parse(it) }
+        val offsetFormat = UtcOffset.Format {
+            optional("Z") {
+                offsetHours()
+                optional {
+                    char(':'); offsetMinutesOfHour()
+                    optional { char(':'); offsetSecondsOfMinute() }
+                }
+            }
+        }
+        val offsets = offsetStrings.map { UtcOffset.parse(it, offsetFormat) }
```

Comment:
Kinda strange that `DateTimeComponents.Formats.ISO_DATE_TIME_OFFSET` cannot be represented as a combination of `ISO_DATE_TIME` and `ISO_OFFSET`